<launch>
    <!-- define parameters -->
    <arg name="tf_prefix" default=""/>
    <arg name="sim" default="false" description="true: activates simulation time and might load different parameters"/>
   
    <let if="$(env IS_ROBOT false)" name="taskset" value="taskset -c 8"/>
    <let unless="$(env IS_ROBOT false)" name="taskset" value=""/>
    <let if="$(env IS_ROBOT false)" name="handler_taskset" value="taskset -c 9"/>
    <let unless="$(env IS_ROBOT false)" name="handler_taskset" value=""/>

    <!-- load localization -->
    <node pkg="bitbots_localization" exec="localization" name="bitbots_localization" output="screen" launch-prefix="$(var taskset)">
        <!-- load base parameters -->
        <param from="$(find-pkg-share bitbots_localization)/config/config.yaml"/>
        <!-- load field specific paramters -->
        <param from="$(find-pkg-share bitbots_localization)/config/fields/$(var fieldname)/config.yaml"/>

        <param name="field.name" value="$(var fieldname)"/>
        <!-- load correct frames for namespace -->
        <param name="ros.odom_frame" value="$(var tf_prefix)odom"/>
        <param name="ros.base_footprint_frame" value="$(var tf_prefix)base_footprint"/>
        <param name="ros.map_frame" value="$(var tf_prefix)map"/>
        <param name="ros.publishing_frame" value="$(var tf_prefix)localization_raw"/>
        <param name="use_sim_time" value="$(var sim)"/>
    </node>

    <node pkg="bitbots_localization_handler" exec="localization_handler" output="screen" launch-prefix="$(var handler_taskset)">
        <param name="use_sim_time" value="$(var sim)"/>
        <param name="odom_frame" value="$(var tf_prefix)odom"/>
        <param name="base_footprint_frame" value="$(var tf_prefix)base_footprint"/>
        <param name="walking_moved_distance" value="0.5"/>
    </node>
</launch>
